pageserver/src/layered_repository/delta_layer.rs

```diff
@@ -420,6 +420,21 @@ impl DeltaLayer {
         }
     }
 
+    fn temp_path_for(
+        conf: &PageServerConf,
+        timelineid: ZTimelineId,
+        tenantid: ZTenantId,
+        key_start: Key,
+        lsn_range: Range<Lsn>,
+    ) -> PathBuf {
+        conf.timeline_path(&timelineid, &tenantid).join(format!(
+            "{}-XXX__{:016X}-{:016X}.temp",
```

Comment:
We can also resolve https://github.com/neondatabase/neon/issues/1739 in this PR by appending a random string to the temp file name.

This, IMHO, is similar to how many temp file generators work.

Reply:
Thank you for pointing out.
Fixed.